service/device/registry/internal/src/main/java/org/eclipse/kapua/service/device/registry/connection/option/internal/DeviceConnectionOptionServiceImpl.java

```diff
@@ -62,10 +69,23 @@ public DeviceConnectionOption update(DeviceConnectionOption deviceConnectionOpti
         //
         // Check Access
         KapuaLocator locator = KapuaLocator.getInstance();
+        DeviceConnectionService deviceConnectionService = locator.getService(DeviceConnectionService.class);
         AuthorizationService authorizationService = locator.getService(AuthorizationService.class);
         PermissionFactory permissionFactory = locator.getFactory(PermissionFactory.class);
         authorizationService.checkPermission(permissionFactory.newPermission(DeviceDomains.DEVICE_CONNECTION_DOMAIN, Actions.write, deviceConnectionOptions.getScopeId()));
 
+        if (deviceConnectionOptions.getReservedUserId() != null) {
+            AndPredicateImpl deviceAndPredicate = new AndPredicateImpl();
+            deviceAndPredicate.and(new AttributePredicateImpl<>(DeviceConnectionAttributes.SCOPE_ID, deviceConnectionOptions.getScopeId()));
```

Comment:
What's the point of this predicate? I'd say you don't need this since you already create the query for the correct scopeId in line 81